07-routing-gossip.md

```diff
@@ -1105,6 +1106,47 @@ A->D's `update_add_htlc` message would be:
 And D->C's `update_add_htlc` would again be the same as B->C's direct payment
 above.
 
+# Onion Messages
+
+Onion messages allow peers to use existing connections to query for
+invoices (see [BOLT 12](12-offer-encoding.md)).  Like gossip messages,
+they are not associated with a particular local channel.  Like HTLCs,
+they use [BOLT 4](04-onion-routing.md#onion-messages) protocol for
+end-to-end encryption.
+
+Onion messages are unreliable: in particular, they are designed to
+be cheap to process and require no storage to forward.  As a result,
+there is no error return from intermediary nodes.
+
+To enable messaging via blinded paths, there is an optional `blinding`
+parameter which allows decryption of the `enctlv` field inside the
+`onionmsg`'s `onionmsg_payload`.
+
+## The `onion_message` Message
+
+1. type: 385 (`onion_message`) (`option_onion_messages`)
+2. data:
+    * [`1366*byte`:`onionmsg`]
+	* [`onion_message_tlvs`:`onion_message_tlvs`]
```

Comment:
BTW seeing that makes me realize something about the TLV extension in `update_add_htlc`.
For blinded paths, the `update_add_htlc` will have a different size from "normal" `update_add_htlc`s.
That means their anonymity set will be reduced and network observers could unblind paths.

I don't have a solution yet, but we should try to make all `update_add_htlc`s the same size, otherwise it's somewhat useless to have a fixed-size onion in it...

Reply:
You can pad with a no-reply ping for this?  Have to make sure the numbers are right though!